x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportReRunAction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ *
+ */
+package org.elasticsearch.xpack.indexlifecycle.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.AckedClusterStateUpdateTask;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.indexlifecycle.ErrorStep;
+import org.elasticsearch.xpack.core.indexlifecycle.LifecycleSettings;
+import org.elasticsearch.xpack.core.indexlifecycle.Step.StepKey;
+import org.elasticsearch.xpack.core.indexlifecycle.action.ReRunAction;
+import org.elasticsearch.xpack.core.indexlifecycle.action.ReRunAction.Request;
+import org.elasticsearch.xpack.core.indexlifecycle.action.ReRunAction.Response;
+import org.elasticsearch.xpack.indexlifecycle.IndexLifecycleRunner;
+import org.elasticsearch.xpack.indexlifecycle.IndexLifecycleService;
+
+public class TransportReRunAction extends TransportMasterNodeAction<Request, Response> {
+    IndexLifecycleService indexLifecycleService;
+    @Inject
+    public TransportReRunAction(Settings settings, TransportService transportService, ClusterService clusterService,
+                                ThreadPool threadPool, ActionFilters actionFilters,
+                                IndexNameExpressionResolver indexNameExpressionResolver,
+                                IndexLifecycleService indexLifecycleService) {
+        super(settings, ReRunAction.NAME, transportService, clusterService, threadPool, actionFilters, indexNameExpressionResolver,
+                Request::new);
+        this.indexLifecycleService = indexLifecycleService;
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected Response newResponse() {
+        return new Response();
+    }
+
+    ClusterState innerExecute(ClusterState currentState, String[] indices) {
```

Comment:
Can we put this method in IndexLifecycleRunner to keep all the step transition logic in the same place?

Reply:
yes